src/core/utils/parse_cmdline.cpp

```diff
@@ -264,6 +264,31 @@ Error ParseAsIp6Address(const char *aString, otIp6Address &aAddress)
     return (aString != nullptr) ? otIp6AddressFromString(aString, &aAddress) : kErrorInvalidArgs;
 }
 
+Error ParseAsIp4Address(const char *aString, otIp4Address &aAddress)
+{
+    return (aString != nullptr) ? otIp4AddressFromString(aString, &aAddress) : kErrorInvalidArgs;
+}
+
+Error ParseAndConvertToIp6Address(otInstance *aInstance, const char *aString, otIp6Address &aAddress, bool &aConverted)
+{
+    Error error = kErrorNone;
+
+    VerifyOrExit(aString != nullptr, error = kErrorInvalidArgs);
+    error      = ParseAsIp6Address(aString, aAddress);
+    aConverted = false;
+    if (error != kErrorNone)
+    {
+        // It might be an IPv4 address, let's have a try.
+        otIp4Address ip4Address;
+
+        SuccessOrExit(otIp4AddressFromString(aString, &ip4Address));
+        SuccessOrExit(error = otNat64SynthersizeIp6Address(aInstance, &ip4Address, &aAddress));
+        aConverted = true;
+    }
+exit:
+    return error;
```

Comment:
Thanks for adding this common helper.

One nit/suggestion about this:
- I suggest adding `ParseAndConvertToIp6Address()` in `Cli::Interpreter` instead of in `parse_cmdline` method.
- The idea is that `parse_cmdline` provides simple helper parsing functions (parse strings/inputs into different simple types)
- So adding `ParseAsIp4Address()` make sense to me
- But this is more complex logic and is doing NAT64 address conversion
- I feel this may not fit in `parse_cmdline` module and may be better if we move this in `CLI` module.
- We already have some CLI specific `Parse` methods defined in `cli.cpp` (e.g. `ParsePreference`, `ParsePingInterval` or `ParseJoinerDiscerner`, ...)

Reply:
Updated, thanks for advice.